invenio_admin/ext.py

```diff
@@ -40,49 +72,42 @@ class InvenioAdmin(object):
     :param app: Flask application.
     :param entry_point_group: Name of entry point group to load views/models
         from.
+    :param permission_factory: Default permission factory to use when
+        protecting admin view.
+    :param viewcls_factory: Factory for creating admin view classes on the
+        fly. Used to protect admin views with authentication and authorization.
+    :param indeview_cls: Admin index view class.
     """
 
-    def __init__(self, app=None, anonymous=False, **kwargs):
-        """InvenioAdmin extension initialization.
-
-        If `anonymous` is True, the admin views are accessible to anonymous
-        users and all security checks are bypassed (use for testing only).
-        """
-        self.anonymous = anonymous
+    def __init__(self, app=None, **kwargs):
+        """InvenioAdmin extension initialization."""
         if app:
-            self.init_app(app, **kwargs)
+            self._state = self.init_app(app, **kwargs)
 
-    def init_app(self, app, entry_point_group='invenio_admin.views',
+    def init_app(self,
+                 app,
+                 entry_point_group='invenio_admin.views',
+                 permission_factory=admin_permission_factory,
+                 viewcls_factory=protected_adminview_factory,
```

Comment:
can we make the argument names more uniform?

#### Proposals
- `view_factory`, `view_class_factory`
- `index_view_class`, `index_view`